loadtest/loadtest.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package loadtest
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/mattermost/mattermost-load-test-ng/config"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/control"
+
+	"github.com/mattermost/mattermost-server/v5/mlog"
+)
+
+// LoadTester is a structure holding all the state needed to run a load-test
+type LoadTester struct {
+	controllers   []control.UserController
+	config        *config.LoadTestConfig
+	wg            sync.WaitGroup
+	status        chan control.UserStatus
+	started       bool
+	newController NewController
+}
+
+// NewController is a factory function that returns a new
+// control.UserController given an id and a channel of control.UserStatus
+type NewController func(int, chan<- control.UserStatus) control.UserController
+
+func (lt *LoadTester) handleStatus() {
+	for us := range lt.status {
+		if us.Code == control.USER_STATUS_STOPPED || us.Code == control.USER_STATUS_FAILED {
+			lt.wg.Done()
+		}
+		if us.Code == control.USER_STATUS_ERROR {
+			mlog.Info(us.Err.Error(), mlog.Int("controller_id", us.ControllerId))
```

Comment:
Should this be `mlog.Error` ?

Reply:
I think originally the idea was to treat errors related to users as normal information since they are expected to happen but it's fine either way as long as we can track everything from the logs.